### PR TITLE
Nil loader no panic, only load project if loader is not nil when get provider

### PR DESCRIPTION
**Pull request by edwardrf**
## Description
cli panics when non-interactively executed and reaching quota.

## Checklist

- [X] I have performed a self-review of my code
- [x] I have added appropriate tests
- [ ] I have updated the Defang CLI docs and/or README to reflect my changes, if necessary

### CODE REVIEW

File: src/pkg/cli/compose/loader.go [hunk 1]
```diff
@@ -67,6 +67,9 @@ func NewLoader(opts ...LoaderOption) *Loader {
 }
 
 func (c *Loader) LoadProjectName(ctx context.Context) (string, error) {
+	if c == nil {
+		return "", errors.New("missing loader")
```
Comment (lionello) [inline src/pkg/cli/compose/loader.go:71]: will the user see this? Means nothing to them. Might as well leave panic.

Comment (edwardrf) [reply src/pkg/cli/compose/loader.go]: Done

File: src/cmd/cli/command/commands.go [hunk 2]
```diff
@@ -1256,17 +1259,20 @@ func getProvider(ctx context.Context, loader *compose.Loader) (cliClient.Provide
...
-			term.Warn("Unable to get selected provider:", err)
-		} else if resp.Provider != defangv1.Provider_PROVIDER_UNSPECIFIED {
-			providerID.SetEnumValue(resp.Provider)
-			return "stored preference", nil
+func determineProviderID(ctx context.Context, loader cliClient.Loader) (string, error) {
+	var projName string
+	if loader != nil {
+		if name, err := loader.LoadProjectName(ctx); err != nil {
+			term.Warn("Unable to load project:", err)
+		} else if name != "" && !RootCmd.PersistentFlags().Changed("provider") { // If user manually selected auto provider, do not load from remote
```
[the remark this answers is not in the document]
Comment (edwardrf) [reply src/cmd/cli/command/commands.go:1267]: This is the logic where user use `--provider=auto` to interactively re-select provider

File: src/pkg/cli/client/grpc.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ import (
 
 type GrpcClient struct {
 	anonID string
-	client defangv1connect.FabricControllerClient
+	Client defangv1connect.FabricControllerClient
```
[the remark this answers is not in the document]
Comment (edwardrf) [reply src/pkg/cli/client/grpc.go]: Yes, i changed it back and added a SetClient method.

File: src/pkg/cli/compose/loader_test.go [hunk 4]
```diff
@@ -126,3 +126,14 @@ func TestHasSubstitution(t *testing.T) {
...
+func TestNilLoader(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("Nil loader should panic when loading project")
+		}
+	}()
+	var loader *Loader
+	loader.LoadProject(context.Background())
+	t.Error("Should not execute beyond nil loader.LoadProject()")
+}
```
Comment (lionello) [inline src/pkg/cli/compose/loader_test.go:139]: This is the kind of test that's written after the fact. Nobody would say "I want this to panic" and then implement the panic.
```suggestion
```

**Review by lionello — APPROVED** [review]

**Pull request merged**